airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -134,8 +134,12 @@ def _evaluate_trigger_rule(  # pylint: disable=too-many-branches
                 if successes or skipped:
                     ti.set_state(State.SKIPPED, session)
             elif trigger_rule == TR.ONE_SUCCESS:
-                if upstream_done and not successes:
+                if upstream_done and done == skipped:
+                    # if upstream is done and all are skipped mark as skipped
                     ti.set_state(State.SKIPPED, session)
+                elif upstream_done and successes <= 0:
```

Comment:
How can successes be less than zero?

Reply:
Unclear to me; I'm just following the pattern I see below so that this is consistent.

https://github.com/apache/airflow/blob/5e79b1ed7551345c8b2ffebbef6c873e98058a14/airflow/ti_deps/deps/trigger_rule_dep.py#L155

Reply:
Fair